Interop/ModelCard/src/main/java/ModelDetails.java

```diff
@@ -0,0 +1,115 @@
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.oracle.labs.mlrg.olcut.provenance.ConfiguredObjectProvenance;
+import org.tribuo.Model;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ModelDetails {
+    private static final ObjectMapper mapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);;
+    private final String schemaVersion;
+    private final String modelType;
+    private final String modelPackage;
+    private final String tribuoVersion;
+    private final String javaVersion;
+    private final Map<String, Object> configuredParams = new HashMap<>();
+
+    public ModelDetails(Model<?> model) {
+        schemaVersion = "1.0";
+        modelType = model.getClass().getSimpleName();
+        modelPackage = model.getClass().getTypeName();
+        tribuoVersion = model.getProvenance().getTribuoVersion();
+        javaVersion = model.getProvenance().getJavaVersion();
+
+        Map<String,?> parameters = model.getProvenance().getTrainerProvenance().getConfiguredParameters();
+        for (String key : parameters.keySet())
+            configuredParams.put(key, parameters.get(key));
+    }
+
+    public ModelDetails(JsonNode modelDetailsJson) throws JsonProcessingException {
```

Comment:
My reasoning was that because a user has the capability to make a specific chunk of a ModelCard (e.g. just the UsageDetails) and save that to file, they might then later need to also deserialize it too. I've kept the constructor public for now, but if you still think it's best to have it be package private, I'm more than happy to change it.

Reply:
My reasoning was that because a user has the capability to make a specific chunk of a ModelCard (e.g. just the ModelDetails) and save that to file, they might then later need to also deserialize it too. I've kept the constructor public for now, but if you still think it's best to have it be package private, I'm more than happy to change it.